sample/rules.go

```diff
@@ -190,11 +199,35 @@ func ruleMatchesSpanInTrace(trace *types.Trace, rule *config.RulesBasedSamplerRu
 		return true
 	}
 
+	var seen []int
+
 	for _, span := range trace.GetSpans() {
 		ruleMatched := true
-		for _, condition := range rule.Conditions {
-			// whether this condition is matched by this span.
-			value, exists := extractValueFromSpan(trace, span, condition, checkNestedFields)
+		for i, condition := range rule.Conditions {
+			var value interface{}
+			var exists bool
+			if slices.Contains(seen, i) {
+				continue
+			} else {
+				if strings.Contains(condition.Field, RootPrefix) {
```

Comment:
Validation checks that you don't define both Field and Fields for a given rule, and during the Init() function, if it's Field we just copy it into Fields[] as the only element, so that we never have to check anything but Fields. You can eliminate this condition entirely.

Reply:
Great, I'll remove it.